crates/oxc_semantic/src/builder.rs

```diff
@@ -1633,6 +1633,15 @@ impl<'a> Visit<'a> for SemanticBuilder<'a> {
             self.visit_function_body(body);
         }
 
+        control_flow!(self, |cfg| {
+            let c = cfg.current_basic_block();
+            if !matches!(
```

Comment:
Shall we use `if cfg.current_basic_block().unreachable` to detect more complex cases where somewhere up the graph we have returned?

Example:

```js
function f() {
  try {
    return g(x);
  } catch {
    return f();
  }
}
```

<details>
<summary>Before:</summary>
![image](https://github.com/user-attachments/assets/fd3a2dfb-aac1-4a3b-bcd3-64564ce668a6)
</details>

<details>
<summary>After:</summary>
![image](https://github.com/user-attachments/assets/0d0743ef-6e2b-452e-bd80-749834e785f5)
</details>

Biome has the redundant return block but then again they always add a return block at the end, If I'm not mistaken we want to have a return and don't care for a dedicated block for it(as long as we encounter a return instruction in our path to the end of the function).

Please let me know what you think.

Reply:
> Shall we use if cfg.current_basic_block().unreachable to detect more complex cases where somewhere up the graph we have returned?

I don't think we need to add a `unreachable` after a `implicit return`, because a `implicit return` must be last instruction of a function other than an `explicit return`

Reply:
what about  rename `ImplicitReturn` to `DummyReturn`  and don't insert a extra `unreachable`,

Reply:
Now that I'm thinking about this, Maybe we can remove the unreachable instruction altogether, It was kept from the old implementation but I've added this field later on to cache some parts of the analysis from the build phase.